### DISCUSSION TITLE
After build dynamic routes are throwing 404.

### DISCUSSION THREAD

**Discussion by singhpriyakr** [vercel/next.js#13073] [Help]
Every thing was working fine on local development, once my app done I run the `next build ` once it done and when I tried build on browser most of the dynamic routes were throwing 404 and that was very weird. 

## To Reproduce

- Have some dynamic page (multi level), 
- Run `next build `

## Expected behavior

- Should not throw 404 for dynamic routes.

## Screenshots

**Screen short of the build output**
<img width="669" alt="Screenshot 2020-05-19 at 8 19 28 AM" src="https://user-images.githubusercontent.com/12538294/82279176-80f0aa00-99a9-11ea-9ded-407c22eaa130.png">
**Screen short of the 404 output**
<img width="1315" alt="Screenshot 2020-05-19 at 8 20 59 AM" src="https://user-images.githubusercontent.com/12538294/82279260-b8f7ed00-99a9-11ea-99c5-29f93ef29ee6.png">
**Screen short of how I am using link tag**
<img width="1196" alt="Screenshot 2020-05-19 at 8 21 55 AM" src="https://user-images.githubusercontent.com/12538294/82279299-d462f800-99a9-11ea-8e27-d4413a391da3.png">
**Screen short of how I have the folder structure**
<img width="228" alt="Screenshot 2020-05-19 at 8 23 05 AM" src="https://user-images.githubusercontent.com/12538294/82279359-feb4b580-99a9-11ea-9b2a-435a4211e04e.png">

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

## Additional context

Even tried changing sequence of `href` and `as`, also tried remove prefetch and adding but output remains same. Is't it very strange, even though you are doing everything right but getting 404 in browsers for all the links. Sad :(

#Update 

I tired removing <Link> tag and tried to handle via Router.push 
eg:

`openCouponPage = (url) => {
	Router.push('/coupon/[id]', url);
}`

But this is also not working.

**Comment by srikanth912** [comment]
In your first Link, try replacing href with actual page (like your 3rd Link) and also use passHref. 

For example:

`<Link href="/[sectionSlug]/[categorySlug]" as={`/${category.section.slug}/${category.parent.slug}`} passHref><a>{category.parent.name}</a></Link>`
